tensorflow-examples/src/main/java/org/tensorflow/model/examples/tensors/TensorCreation.java

```diff
@@ -0,0 +1,62 @@
+package org.tensorflow.model.examples.tensors;
+
+import org.tensorflow.Tensor;
+import org.tensorflow.tools.Shape;
+import org.tensorflow.tools.ndarray.IntNdArray;
+import org.tensorflow.tools.ndarray.NdArrays;
+import org.tensorflow.types.TInt32;
+
+/**
+ * Creates a few tensors of ranks: 0, 1, 2, 3.
+ */
+public class TensorCreation {
+    public static void main(String[] args) {
+        // Rank 0 Tensor
+        Tensor<TInt32> rank0Tensor = TInt32.scalarOf(42);
+        System.out.println("---- Scalar tensor ---------");
+        System.out.println("DataType: " + rank0Tensor.dataType().name());
+        System.out.println("Rank: " + rank0Tensor.data().rank());
+        System.out.println("NumElements: " + rank0Tensor.data().size());
```

Comment:
I think a user should access `data()` only when he's interested in doing some read/write operations to the tensor memory, as it might move the tensor from some remote location or from GPU memory to the local CPU for this purpose.

Therefore, if the user intent is only to display some metadata about the shape of the tensor, the `tensor.shape()` method is probably better. (`rank0Tensor.shape().numDimensions()` and `rank0Tensor.shape().size()` respectively in this case).

Reply:
Agree, but better to have rank() method in tensor interface for next different logic handling for tensors with the different rank

Agree, that maybe I should call methods on data() only for read/write purposes